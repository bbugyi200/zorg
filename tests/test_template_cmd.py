"""Tests for the zorg project's 'template' CLI command."""

from __future__ import annotations

from pathlib import Path

from _pytest.capture import CaptureFixture
from syrupy.assertion import SnapshotAssertion as Snapshot

from . import common as c


def test_template_render_day_log(
    main: c.MainType,
    capsys: CaptureFixture,
    tmp_path: Path,
    snapshot: Snapshot,
) -> None:
    """Test that the 'day_log_tmpl.zo' template renders as expected."""
    zettel_dir = tmp_path / "org"
    zot_basename = "day_log_tmpl.zo"
    template_path: Path = zettel_dir / zot_basename
    template_path.parent.mkdir(parents=True, exist_ok=True)
    test_data_template_path = Path(__file__).parent / Path(
        f"data/{zot_basename}"
    )
    template_path.write_text(test_data_template_path.read_text())

    argv = [
        "--dir",
        str(zettel_dir),
        "template",
        "render",
        zot_basename,
        "date=19910304",
    ]
    exit_code = main(*argv)

    assert exit_code == 0
    capture = capsys.readouterr()
    assert capture.out == snapshot


def test_template_render_hello(
    main: c.MainType,
    capsys: CaptureFixture,
    tmp_path: Path,
) -> None:
    """Test that the 'hello_tmpl.zo' template renders as expected."""
    zettel_dir = tmp_path / "org"
    zot_basename = "hello_tmpl.zo"
    template_path: Path = zettel_dir / zot_basename
    template_path.parent.mkdir(parents=True, exist_ok=True)
    test_data_template_path = Path(__file__).parent / Path(
        f"data/{zot_basename}"
    )
    template_path.write_text(test_data_template_path.read_text())

    argv = [
        "--dir",
        str(zettel_dir),
        "template",
        "render",
        zot_basename,
        "first_name=John",
        "last_name=Smith",
    ]
    assert main(*argv) == 0

    capture = capsys.readouterr()
    assert capture.out == "Hello John Smith\n"
