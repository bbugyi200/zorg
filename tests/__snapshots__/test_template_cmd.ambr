# serializer version: 1
# name: test_template_render_day_log[YAMLConfigFile]
  '''
  # 1991-03-04.Mon
  #
  # @ = [[day_log_tmpl]]
  # < = [[19910303.zo]]
  # > = [[19910305.zo]]
  #
  # H = [[19910304_habit.zo]]
  # X = [[19910304_done.zo]]
  
  o #gtd pomodoro
    * Process yesterday's bujo log (e.g. close events, add missing pomodoros, rollover goals)
    * Copy any important gchat convos from yesterday into chat/*.txt
    * Google Calendar (bbugyi@google + bryanbugyi34@gmail)
    * Check today's tickler file: [[tick_4]]
    * Fill out yesterday's habit tracker: [[19910303_habit.zo]]
    * go/g3co
    * Review '@A' todos (are any >=7d old? which should you do today?)
    * Set 2-5 goals / "highlights" for today
    * Add [[act]] file links to this day log
    * <50 emails in bryanbugyi34@gmail
    * <100 emails in bbugyi@google
    * inbox: Google Keep
    * inbox: ~/org/box.zo
    * Review ~/org diffs from yesterday and commit them to git
  
  ##### Goals #####
  
  ##### Events #####
  
  ##### Pomodoros #####
  
  '''
# ---
